packages/pds/src/api/app/bsky/feed/getPostThread.ts

```diff
@@ -22,25 +22,41 @@ import {
   getRepoRev,
   handleReadAfterWrite,
 } from '../util/read-after-write'
+import { authPassthru } from '../../../../../api/com/atproto/admin/util'
 
 export default function (server: Server, ctx: AppContext) {
   server.app.bsky.feed.getPostThread({
-    auth: ctx.accessVerifier,
-    handler: async ({ params, auth }) => {
-      const requester = auth.credentials.did
+    auth: ctx.accessOrRoleVerifier,
+    handler: async ({ req, params, auth }) => {
+      const requester =
+        auth.credentials.type === 'access' ? auth.credentials.did : null
       try {
         const res = await ctx.appviewAgent.api.app.bsky.feed.getPostThread(
           params,
-          await ctx.serviceAuthHeaders(requester),
-        )
-        return await handleReadAfterWrite(
-          ctx,
-          requester,
-          res,
-          getPostThreadMunge,
+          requester
+            ? await ctx.serviceAuthHeaders(requester)
+            : authPassthru(req),
         )
+
+        if (requester) {
+          return await handleReadAfterWrite(
+            ctx,
+            requester,
+            res,
+            getPostThreadMunge,
+          )
+        }
+
+        return {
+          encoding: 'application/json',
+          body: res.data,
+        }
       } catch (err) {
-        if (err instanceof AppBskyFeedGetPostThread.NotFoundError) {
+        // TODO: is this right? checking for requester and error type is safe enough?
+        if (
+          err instanceof AppBskyFeedGetPostThread.NotFoundError &&
+          requester
+        ) {
```

Comment:
Yeah, I believe that should be fine 👍 Maybe there's a way to split the two cases-up more rather than have the `requester` checks sprinkled throughout.  For example, maybe the case could be offloaded up front since it's relatively simple on its own:

```ts
      const requester =
        auth.credentials.type === 'access' ? auth.credentials.did : null

      if (!requester) {
        const res = await ctx.appviewAgent.api.app.bsky.feed.getPostThread(
          params,
          authPassthru(req),
        )
        return {
          encoding: 'application/json',
          body: res.data,
        }
      }

      // ...
      // rest of code continues as originally written
      // ...
```

Reply:
ah yes, looks much better. just wasn't sure if the whole `handleReadAfterWrite` is needed for mods too so didn't spend any time thinking about the implementation. cleaned it up as suggested, thanks!